docs/deployment.md

```diff
@@ -0,0 +1,483 @@
+# Create the KubeVirt CR (instance deployment request) which triggers the actual installation
+kubectl apply --filename https://github.com/kubevirt/kubevirt/releases/download/${RELEASE}/kubevirt-cr.yaml
+# Wait until all KubeVirt components are up
+kubectl --namespace=kubevirt wait kubevirt kubevirt --for condition=Available
+```
+
+# Virt Operator Deployment
+
+`virt-operator` is the component responsible for deploying the components of the KubeVirt project.
+
+## Namespace
+A dedicated [Namespace](https://kubernetes.io/docs/concepts/overview/working-with-objects/namespaces/) is created for KubeVirt's components.
+
+```shell
+kubectl get namespace kubevirt -o yaml
+```
+
+## KubeVirt CustomResourceDefinition
+KubeVirt extends Kubernetes using [CustomResourceDefinitions](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/) (CRDs).
+
+The first deployed CRD is `kubevirts.kubevirt.io` which allows us to later create the `KubeVirt` CustomResource.
+
+```shell
+kubectl get customresourcedefinitions.apiextensions.k8s.io kubevirts.kubevirt.io -o yaml
+```
+
+## Priority Class
+`kubevirt-cluster-critical` [PriorityClass](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass) is deployed and will be used by `virt-operator`.
+
+```shell
+kubectl get priorityclasses.scheduling.k8s.io kubevirt-cluster-critical -o yaml
+```
+
+## ServiceAccount
+A [ServiceAccount](https://kubernetes.io/docs/concepts/security/service-accounts/) is deployed to be used by `virt-operator`.
+
+```shell
+kubectl get --namespace=kubevirt serviceaccount kubevirt-operator -o yaml
+```
+
+## ClusterRoles
+
+```shell
+kubectl get clusterroles.rbac.authorization.k8s.io kubevirt.io:operator -o yaml
+kubectl get clusterroles.rbac.authorization.k8s.io kubevirt-operator -o yaml
+```
+
+## ClusterRoleBinding
+
+Binds the `kubevirt-operator` ClusterRole to the `kubevirt:kubevirt-operator` ServiceAccount.
+
+```shell
+kubectl get clusterrolebindings.rbac.authorization.k8s.io kubevirt-operator -o yaml
+```
+
+## Role
+
+```shell
+kubectl get --namespace=kubevirt roles.rbac.authorization.k8s.io kubevirt-operator -o yaml
+```
+
+## RoleBinding
+Binds the `kubevirt-operator` Role to the `kubevirt:kubevirt-operator` ServiceAccount.
+
+```shell
+kubectl get --namespace=kubevirt rolebindings.rbac.authorization.k8s.io kubevirt-operator-rolebinding -o yaml
+```
+
+## Deployment
+A [Deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/) is created to execute `virt-operator`.
+
+```shell
+kubectl get --namespace=kubevirt deployments.apps virt-operator -o yaml
+```
+
+## KubeVirt CustomResource
+
+After the `virt-operator` and its dependencies are deployed, we deploy the KubeVirt CustomResource in order to trigger virt-operator to reconcile it and deploy KubeVirt's main components.
+
+```shell
+kubectl get --namespace=kubevirt kubevirts.kubevirt.io kubevirt -o yaml
+```
+
+# Operands
```

Comment:
Wow.

How will you ensure that all the stuff below will be kept up to date?

Reply:
The reasons for updating this document could be:
1. Renaming one or more objects.
2. Addition or removal of objects (such as a CRD).

The document intentionally does not contain any details about the objects and instructs the user to fetch the details by name using the `kubectl` utility.

There is a checkbox in the PR's TODO list, it currently refers to the user-guide but could also refer to the developer's guide.
WDYT?